File: 프로그래머스/4/140105. 쌍둥이 빌딩 숲/쌍둥이 빌딩 숲.js
const DIVIDER = 1000000007;

function solution(n, count) {
    const dp = Array.from({length: n + 1}, () => Array.from({length: n + 1}, () => 0))
    dp[1][1] = 1;
    for(let i = 2; i <= n; i++) {
        for(let j = 1; j <= i; j++) {
            dp[i][j] = ((dp[i - 1][j - 1] || 0) + dp[i - 1][j] * 2 * (i - 1)) % DIVIDER;
        }
    }
    return dp[n][count];
}

File: 프로그래머스/4/140105. 쌍둥이 빌딩 숲/쌍둥이 빌딩 숲.py
DIVIDER = 1000000007

def solution(n, count):
    dp = [[0 for _ in range(n + 1)] for _ in range(n + 1)]
    dp[1][1] = 1
    for i in range(2, n + 1):
        for j in range(1, i + 1):
            placeFront = dp[i - 1][j - 1] if j > 1 else 0
            placeBack = dp[i - 1][j] * 2 * (i - 1)
            dp[i][j] = (placeFront + placeBack) % DIVIDER
    return dp[n][count]
